httpclient5/src/main/java/org/apache/hc/client5/http/entity/DecompressEntity.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * ====================================================================
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ * ====================================================================
+ *
+ * This software consists of voluntary contributions made by many
+ * individuals on behalf of the Apache Software Foundation.  For more
+ * information on the Apache Software Foundation, please see
+ * <http://www.apache.org/>.
+ *
+ */
+package org.apache.hc.client5.http.entity;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.hc.core5.annotation.Contract;
+import org.apache.hc.core5.annotation.ThreadingBehavior;
+import org.apache.hc.core5.http.HttpEntity;
+import org.apache.hc.core5.http.io.entity.HttpEntityWrapper;
+import org.apache.hc.core5.util.Args;
+
+/**
+ * An {@link HttpEntity} wrapper that decompresses the content of the wrapped entity.
+ * This class supports different compression types and can handle both standard
+ * compression (e.g., gzip, deflate) and variations that require a custom handling (e.g., noWrap).
+ *
+ * <p>Decompression is performed using a {@link LazyDecompressInputStream} that
+ * applies decompression lazily when content is requested.</p>
+ *
+ * <p>
+ * Note: This class uses lazy initialization for certain fields, making it <b>not thread-safe</b>.
+ * If multiple threads access an instance of this class concurrently, they must synchronize on the instance
+ * to ensure correct behavior.
+ * </p>
+ *
+ * @since 5.5
+ */
+@Contract(threading = ThreadingBehavior.UNSAFE)
+public class DecompressEntity extends HttpEntityWrapper {
```

Comment:
The class name is mismatched with the "CompressingEntity" name, either they should both be "...ing" names, or not.